### PR TITLE
Refactor client.OpenSergoClient with new model and refine code

### LINKED ISSUE #14: Improve error handling in client.NewOpenSergoClient()
## Issue Description

Type: *enhancement*

### Describe what feature you want

Improve error handling in `client.NewOpenSergoClient(host, port)`. The error should not be ignored.

### PR DESCRIPTION
**Pull request by sczyh30**
### Describe what this PR does / why we need it

Refactor client.OpenSergoClient with new model and refine code.

A sample of usage:

```go
err := client.SubscribeConfig(subscribeKey, api.WithSubscriber(s1), api.WithSubscriber(s2))
```

### Does this pull request fix one issue?

Resolves #14, #15

### Describe how you did it

* Rearrange `SubscribeKey` to `model` pkg and refine `SubscriberRegistry`
* Add `api.OpenSergoClient` interface and functional SubscribeOptions
* Refactor client.OpenSergoClient with the new model and improve error handling
* Update samples

### Describe how to verify it

Test cases TBD...

### Special notes for reviews

NOTE: this PR contains breaking changes of API.

### CODE REVIEW

**Review by jnan806 — APPROVED** [review]

File: samples/main/main.go [hunk 1]
```diff
@@ -31,43 +33,46 @@ func main() {
...
-	sampleFaultToleranceRuleSubscriber := new(samples.SampleFaultToleranceRuleSubscriber)
-	// 3. construct SubscribeInfo
-	faultToleranceSubscribeInfo := client.NewSubscribeInfo(faultToleranceSubscribeKey)
-	faultToleranceSubscribeInfo.AppendSubscriber(sampleFaultToleranceRuleSubscriber)
-	// 4. registry
-	openSergoClient.RegisterSubscribeInfo(faultToleranceSubscribeInfo)
+	// Create a OpenSergoClient.
+	openSergoClient, err := client.NewOpenSergoClient("127.0.0.1", 10246)
+	if err != nil {
+		log.Fatalln("Failed to create OpenSergoClient: " + err.Error())
```
Comment (jnan806) [inline samples/main/main.go:41]: `log.Fataln()` will invoke `os.Exit(1)`, the hole application will be shutdown. May we wait for keeping alive ? I think  only log the messages and left a `// TODO` to handle by user self. Just like follwing:
``` go
logging.Error("Failed to create OpenSergoClient: " + err.Error())
// TODO the error logic
```

File: samples/main/main.go [hunk 2]
```diff
@@ -31,43 +33,46 @@ func main() {
...
+	}
 
-	// start OpensergoClient
-	openSergoClient.Start()
+	// Start OpenSergoClient
+	err = openSergoClient.Start()
+	if err != nil {
+		log.Fatalln("Failed to start OpenSergoClient: " + err.Error())
```
Comment (jnan806) [inline samples/main/main.go:47]: the same as above.

File: samples/main/main.go [hunk 3]
```diff
@@ -31,43 +33,46 @@ func main() {
...
-	rateLimitSubscribeInfo.AppendSubscriber(sampleRateLimitStrategySubscriber)
-	openSergoClient.RegisterSubscribeInfo(rateLimitSubscribeInfo)
+	// Create a SubscribeKey for FaultToleranceRule.
+	faultToleranceSubscribeKey := model.NewSubscribeKey("default", "foo-app", configkind.ConfigKindRefFaultToleranceRule{})
+	// Create a Subscriber.
+	sampleFaultToleranceRuleSubscriber := &samples.SampleFaultToleranceRuleSubscriber{}
+	// Subscribe data with the key and subscriber.
+	err = openSergoClient.SubscribeConfig(*faultToleranceSubscribeKey, api.WithSubscriber(sampleFaultToleranceRuleSubscriber))
+	if err != nil {
+		log.Fatalln("Failed to SubscribeConfig: " + err.Error())
```
Comment (jnan806) [inline samples/main/main.go:57]: the same as above

**Review by jnan806 — CHANGES_REQUESTED** [review]

File: samples/main/main.go (hunk 1 above)
Comment (sczyh30) [reply samples/main/main.go:41]: Actually some of the errors are not recoverable. The handling logic depends on developers (thinking it fatal or not). 

When `Start` the client, it's indeed weird to return an error  but has a keep-alive loop internal (and the client may finally recover...). We may need to improve the auto-keep-alive model and error model of the client to make it more clear. We may discuss this in a new issue.

Comment (sczyh30) [reply samples/main/main.go]: I've simplified the sample, but it needs to be updated again once the auto-keep-alive model and error model has been settled down.

**Review by jnan806 — APPROVED** [review]
LGTM

**Pull request merged**